src/App.tsx

```diff
@@ -15,48 +16,113 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  Name = 'name',
+  Length = 'length',
+  Default = '',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortBy,
+  isReversedField: boolean,
+) {
+  const prepareGoods: string[] = [...goods];
+
+  if (sortField) {
+    prepareGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortBy.Name:
+          return good1.localeCompare(good2);
+
+        case SortBy.Length:
+          return good1[sortField] - good2[sortField];
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return isReversedField
+    ? prepareGoods.reverse()
+    : prepareGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortBy.Default);
+  const [isReversedField, setIsReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(
+    goodsFromServer,
+    sortField,
+    isReversedField,
+  );
+
+  const handleResetClicked = () => {
+    setSortField(SortBy.Default);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          onClick={() => setSortField(SortBy.Name)}
+          className={cn(
+            'button',
+            'is-info',
+            { 'is-light': sortField !== SortBy.Name },
+          )}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          onClick={() => setSortField(SortBy.Length)}
+          className={cn(
+            'button',
+            'is-success',
+            { 'is-light': sortField !== SortBy.Length },
+          )}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          onClick={() => setIsReversed(!isReversedField)}
+          className={cn(
+            'button',
+            'is-warning',
+            { 'is-light': !isReversedField },
+          )}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField || isReversedField) && (
```

Comment:
It's better to create a variable for `sortField || isReversedField` so you don't need to combine different logical operators